core/block/editor/basic/details.go

```diff
@@ -66,15 +63,24 @@ func (bs *basic) collectDetailUpdates(details []*pb.RpcObjectSetDetailsDetail, s
 	return updates
 }
 
-// shouldKeepInternalFlags is used to keep internal flags in case we update name or description
-// We keep internal flags because we allow user to change object type and apply some template further
-func shouldKeepInternalFlags(updates []*detailUpdate) bool {
+func removeInternalFlags(s *state.State, updates []*detailUpdate) {
+	flags := internalflag.NewFromState(s.ParentState())
+	if flags.IsEmpty() {
+		return
+	}
+	flags.Remove(model.InternalFlag_editorDeleteEmpty)
+	shouldRemoveFlags := true
 	for _, update := range updates {
 		if update.key == bundle.RelationKeyName.String() || update.key == bundle.RelationKeyDescription.String() {
-			return true
+			shouldRemoveFlags = false
```

Comment:
what if other relations is also changed?

Reply:
Yeah, you are absolutely right. We should keep SelectType and SelectTemplate anyway